spec/components/s3_helper_spec.rb

```diff
@@ -40,17 +46,28 @@
       rules = hash["LifecycleConfiguration"]["Rule"]
 
       expect(rules.length).to eq(2)
-
+      expect(rules[1]["Expiration"]["Days"]).to eq("100")
       # fixes the bad filter
       expect(rules[0]["Filter"]["Prefix"]).to eq("projectdocs/")
     end.to_return(status: 200, body: "", headers: {})
 
-    SiteSetting.enable_s3_uploads = true
-    SiteSetting.s3_access_key_id = "abc"
-    SiteSetting.s3_secret_access_key = "def"
+    helper = S3Helper.new('bob', 'tomb')
+    helper.update_tombstone_lifecycle(100)
+  end
+
+  it "can skip policy update when s3_configure_tombstone_policy is disabled" do
+    SiteSetting.s3_configure_tombstone_policy = false
+
+    stub_request(:get, "https://bob.s3.amazonaws.com/?lifecycle").
+      to_return(status: 200, body: @lifecycle, headers: {})
+
+    hash = Hash.from_xml(@lifecycle)
+    rules = hash["LifecycleConfiguration"]["Rule"]
 
     helper = S3Helper.new('bob', 'tomb')
     helper.update_tombstone_lifecycle(100)
```

Comment:
I've called `update_tombstone_lifecycle()` when `s3_configure_tombstone_policy = false`.
I believe this is what you wanted @SamSaffron